src/_blog/libp2p-webrtc-browser-to-server.md

```diff
@@ -0,0 +1,257 @@
+First, kudos goes to [Little Bear Labs](https://littlebearlabs.io/) who teamed up with Protocol Labs and the libp2p community to define the WebRTC specification and work on the implementation. Protocol Labs authored the [Rust](https://github.com/libp2p/rust-libp2p) implementation, while Little Bear Labs focused on the [Go](https://github.com/libp2p/go-libp2p) and [JavaScript](https://github.com/libp2p/js-libp2p-webrtc) implementations.
+Second, thanks goes to Parity Technologies for helping initiate this effort [many years ago](todo: link to issue or pr) and for all the valuable input on the specification and Rust implementation.
+
+Without further ado, let's begin by introducing WebRTC and how it's currently used. Then we'll go into a deep dive of the WebRTC implementation within libp2p.
+
+## WebRTC in the Browser
+
+`Web Real-Time Communication`, commonly referred to as `WebRTC`, is a [set of standards](https://w3c.github.io/webrtc-pc/) that allows browsers, as well as other clients and servers, to connect to other peers in order to exchange audio, video, and data.  In most cases, peers are directly connected to other peers, allowing for a more private experience and fewer hops than on a relay.
+
+While WebRTC handles audio, video, and data traffic, we're just going to focus on the data aspect because that's the API leveraged in libp2p-webrtc.
+
+WebRTC is built directly into browsers, so using the API is a simple and straightfoward task.  Peers connect to each other via a `RTCPeerConnection` interface.  Once connected, `RTCDataChannels` can be added to the connection to send and receive binary data.
+
+![](https://i.imgur.com/Zv221BT.png)
+
+<!--
+participantspacing 7
+entryspacing 0.6
+
+Peer A->STUN: Who Am I?
+STUN->Peer A: Symmetric Nat
+Peer A->TURN: Channel Please
+Peer A->Signaling Channel: Offer SDP
+Signaling Channel->Peer B: Offer SDP
+Peer B->STUN: Who Am I?
+STUN->Peer B: 159.225.242.189
+Peer B->Signaling Channel: Answer SDP
+Signaling Channel->Peer A: Answer SDP
+
+
+Peer A->(1)Signaling Channel: DTLS Handshake
+Signaling Channel->(1)Peer B: DTLS Handshake
+Peer B->(1)Signaling Channel: DTLS Handshake
+Signaling Channel->(1)Peer A: DTLS Handshake
+
+
+Peer A<->Signaling Channel:Duplex Send/Receive SCTP data, encrypted with DTLS
+Signaling Channel<->Peer B:Duplex Send/Receive SCTP data, encrypted with DTLS
+-->
+
+<div style="font-size: 1.25rem; text-align: center; margin-bottom: 1rem; font-style: italic;">2 browsers connecting via WebRTC, where Peer A has router restrictions (i.e. behind a firewall)</div>
+
+Peers use external [STUN](https://datatracker.ietf.org/doc/html/rfc3489) servers to determine their public address, as well as any router restrictions that prohibit peer-to-peer communications.  In the case of a restriction, [TURN](https://datatracker.ietf.org/doc/html/rfc8656) servers are used to relay data between peers using a Signaling Channel.
```

Comment:
```suggestion
Peers use external [STUN](https://datatracker.ietf.org/doc/html/rfc3489) servers to determine their public address and any router restrictions that prohibit peer-to-peer communications. In the case of a restriction, [TURN](https://datatracker.ietf.org/doc/html/rfc8656) servers relay data between peers using a Signalling Channel.
```

Reply:
Signaling has one L in American English, so I saw we keep it? https://english.stackexchange.com/questions/3946/signalling-or-signaling